### PR TITLE
My Site Dashboard (Phase 2): Quick Action card UI

### LINKED ISSUE #17874: My Site Dashboard (Phase 2): Quick Actions
## Description

Add a Quick Actions card to the dashboard.

Ref: pbArwn-3mP-p2
Design ref: pbArwn-3fp-p2

## Tasks

- [x] Create Quick Actions card UI #17921
- [x] Navigate to the correct screen on tap #17943 
- [x] Track tap events #17943 
- [x] A11y #17921
- [x] Investigate Quick Actions not scrolling (see: p1645562101459479-slack-C0290FLA0RM) #18027

### PR DESCRIPTION
**Pull request by momo-ozawa**
Part of #17874

## Description
- Adds the initial UI for the Quick Actions card on the dashboard

## Notes
- Changes are behind a feature flag
- UI needs refinement -- icons will be added later 
- Quick Actions cards aren't hooked up yet -- tapping won't do anything
- FYI @pachlava -- this feature isn't complete yet

## How to test

1. Replace `BlogDashboardViewModel.swift` L69-74 with this
```
        viewController?.stopLoading()
        applySnapshotWithMockedData()
//        service.fetch(cards: ["posts", "todays_stats"], forBlogID: dotComID, success: { [weak self] _ in
//            self?.viewController?.stopLoading()
//            self?.applySnapshotWithMockedData()
//        }, failure: { _ in
//
//        })
```

2. Add a placeholder icon to `QuickActionsCardCell.swift` on L19
```
        imageView.image = .gridicon(.statsAlt)
```

3. Go to My Site
4. Switch to the dashboard
5. ✅  Quick Actions cards are displayed horizontally and can be scrolled

## Screenshot

Light | Dark
-- | --
 | 

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift [hunk 1]
```diff
@@ -78,15 +76,42 @@ final class BlogDashboardViewController: UIViewController {
 extension BlogDashboardViewController {
 
     private func createLayout() -> UICollectionViewLayout {
-        let layout = UICollectionViewCompositionalLayout {
+        return UICollectionViewCompositionalLayout {
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift]:
```suggestion
       UICollectionViewCompositionalLayout {
```

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewModel.swift [hunk 2]
```diff
@@ -5,18 +5,17 @@ import CoreData
...
-        case quickLinks
+    enum Section: Int, CaseIterable {
+        case quickActions
         case posts
     }
 
     // FIXME: temporary placeholder
-    private let quickLinks = ["Quick Links"]
-    private let posts = ["Posts"]
+    private let quickActions = ["Stats", "Posts", "Media", "Pages"]
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewModel.swift:14]: I know this is a temporary placeholder, but for the subsequent tasks, I think the quick action cell should know about which buttons to display (not this view model).

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewModel.swift [hunk 3]
```diff
@@ -80,14 +81,21 @@ private extension BlogDashboardViewModel {
     // This is necessary when using an IntrinsicCollectionView
     // Otherwise, the collection view will never update its height
     func applySnapshotForInitialData() {
-        let snapshot = Snapshot()
+        var snapshot = Snapshot()
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewModel.swift]: I think you can move that to `applySnapshotWithMockedData`. We should display the entire collection view at once after the request finishes.

**Review by leandroalonso — COMMENTED** [review]
@momo-ozawa there's an issue with Dark mode, this is a change we might want to apply in this PR:

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift (hunk 1 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewController.swift]: Fixed! 3a9e284

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewModel.swift (hunk 3 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Blog/Blog Dashboard/BlogDashboardViewModel.swift]: Fixed! 038798a

**Comment by momo-ozawa** [comment]
> @momo-ozawa there's an issue with Dark mode, this is a change we might want to apply in this PR:
>

@leandroalonso Oops -- thanks for catching this. Fixed! ff958f8
Updated the screenshots section above to include both light / dark mode.

**Comment by jkmassel** [comment]
You can test the <strong>Jetpack</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/jetpack-installable-builds/'>here</a> with build number: <code>pr17908-3a9e284</code>. IPA is available <a href='https://appcenter-filemanagement-distrib2ede6f06e.azureedge.net/d0e7b63f-914e-4397-9392-67729c748a1c/Jetpack%20Alpha.ipa?sv=2019-02-02&sr=c&sig=wvMZ6feV%2BHZxZGvUl96H8xv0mMCiw7bYGaKQQWdiPMo%3D&se=2022-02-09T14%3A00%3A12Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Comment by jkmassel** [comment]
You can test the <strong>WordPress</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/WPiOS-One-Offs/'>here</a> with build number: <code>pr17908-3a9e284</code>. IPA is available <a href='https://appcenter-filemanagement-distrib4ede6f06e.azureedge.net/6d14f5f9-4411-4311-a456-9a27f2adb44b/WordPress%20Alpha.ipa?sv=2019-02-02&sr=c&sig=MjjZrl%2BC4Y6%2BfJPZVmSMdXyoJX3DuBjQ8JqMtWmF%2FLU%3D&se=2022-02-09T14%3A01%3A09Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Comment by pachlava** [comment]
Spent some time trying to test this. I have the code changes applied as mentioned in description, it seems:

Screenshot 2022-02-09 at 14 17 54

Screenshot 2022-02-09 at 14 17 47

And I have what looks like a crash on iPhone Simulator, once I tap `Dashboard` in segmented control.
On iPad, for some reason, I see no segmented control at all (with MSD activated). Both with adjusted code and without.

Most probably, I have something misconfigured. 🤷

**Comment by leandroalonso** [comment]
@pachlava I guess this is because the code in the PRs description was wrong, it should be: `imageView.image = .gridicon(.statsAlt)`.

I had the same issue here (for me the app crashed, though).

**Comment by momo-ozawa** [comment]
@pachlava @leandroalonso Shoot! My apologies for the typo.

As for the iPad -- that's expected for now. We'll be working on showing the dashboard on iPads on [this issue](https://github.com/wordpress-mobile/WordPress-iOS/issues/17878)

Also, just a heads up that I'm going to be closing this PR and re-opening a new one, instead of trying the resolve the conflicts on this PR.

**Comment by momo-ozawa** [comment]
Closing this PR in favor of #17921

**Pull request closed without merging**